lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -340,7 +385,27 @@ function setupEventListeners(transport) {
       transport.disconnect();
       return;
     }
-    transport.disconnect(new SignalingConnectionError());
+    if (transport._reconnectAttemptsLeft <= 0) {
+      transport.disconnect(new SignalingConnectionError());
+    }
+    reconnect();
```

Comment:
Should `connect` and `disconnect` be responsible for removing the "close" and "open" events on the current transport, respectively?

Reply:
@markandrus We are using `.once` and not `.on` for "close" and "open".

Reply:
Right, but if "open" fires, a listener remains for "close" and vice-versa. I don't think we'll have a bug in the underlying TwilioConnection instance, but we'll be accruing listeners nevertheless. I'm not sure if this could lead to a memory leak or not.

Reply:
@markandrus Since we're using an entirely new TwilioConnection upon re-connect, I don't think we'll be accruing listeners. and if the previous TwilioConnection gets garbage-collected, there should be no memory leaks. Let me make sure there are no closure references to it.